contrib/nydus-snapshotter/snapshot/snapshot.go

```diff
@@ -567,60 +567,63 @@ func overlayMount(options []string) []mount.Mount {
 	}
 }
 
+func fuseMount(options []string) []mount.Mount {
+	return []mount.Mount{
+		{
+			Type:    "fuse.nydus-overlayfs",
+			Source:  "overlay",
+			Options: options,
+		},
+	}
+}
+
+type NydusOption struct {
+	Source      string `json:"source"`
+	Config      string `json:"config"`
+	Snapshotdir string `json:"snapshotdir"`
+}
+
 func (o *snapshotter) remoteMounts(ctx context.Context, s storage.Snapshot, id string, labels map[string]string) ([]mount.Mount, error) {
 	var options []string
-	if o.hasDaemon {
-		if s.Kind == snapshots.KindActive {
-			options = append(options,
-				fmt.Sprintf("workdir=%s", o.workPath(s.ID)),
-				fmt.Sprintf("upperdir=%s", o.upperPath(s.ID)),
-			)
-		} else if len(s.ParentIDs) == 1 {
-			return bindMount(o.upperPath(s.ParentIDs[0])), nil
-		}
-		lowerDirOption := fmt.Sprintf("lowerdir=%s", o.upperPath(id))
-		options = append(options, lowerDirOption)
-		log.G(ctx).Infof("mount options %v", options)
-		return overlayMount(options), nil
-	} else {
-		// Only nydus can work without daemon
-		source, err := o.fs.BootstrapFile(id)
-		if err != nil {
-			return nil, err
-		}
-
-		cfg, err := o.fs.NewDaemonConfig(labels)
-		if err != nil {
-			return nil, errors.Wrapf(err, fmt.Sprintf("remoteMounts: failed to generate nydus config for snapshot %s, label: %v", id, labels))
-		}
+	if s.Kind == snapshots.KindActive {
```

Comment:
We need to consider back-compatibility here since some container engine/runtime still depends on `hasDaemon`
mode. So simply removing the if block code snippet is dangerous.

Reply:
agree

Reply:
Fine

Reply:
I find it is difficult to back-compatibility, because when hasDaemon is true, the returned value has changed to a new type  which is  unified value in runc and kata env. Even continue to use hasDaemon, the returned value also has to be changed.  @changweige @bergwolf What do you think?

Reply:
I will consider about this, perhaps tomorrow.
I am quite occupied today. :-(